tests/saispy_ut.cpp

```diff
@@ -0,0 +1,236 @@
+                                    const sai_attribute_t*) {
+        *oid = 1;
+        return (sai_status_t)SAI_STATUS_SUCCESS;
+    };
+
+    acl_api->remove_acl_table = [](sai_object_id_t oid) {
+        return (sai_status_t)(oid == 2 ? SAI_STATUS_SUCCESS : SAI_STATUS_FAILURE);
+    };
+
+    acl_api->set_acl_table_attribute = [](sai_object_id_t oid,
+                                           const sai_attribute_t*) {
+        return (sai_status_t)(oid == 3 ? SAI_STATUS_SUCCESS : SAI_STATUS_FAILURE);
+    };
+
+    acl_api->get_acl_table_attribute = [](sai_object_id_t oid, uint32_t,
+                                           sai_attribute_t*) {
+        return (sai_status_t)(oid == 4 ? SAI_STATUS_SUCCESS : SAI_STATUS_FAILURE);
+    };
+
+    sai_object_id_t oid;
+
+    auto status = acl_api->create_acl_table(&oid, 1, 0, nullptr);
+    ASSERT_TRUE(oid == 1);
+    ASSERT_TRUE(status == SAI_STATUS_SUCCESS);
+
+    status = acl_api->remove_acl_table(2);
+    ASSERT_TRUE(status == SAI_STATUS_SUCCESS);
+
+    status = acl_api->set_acl_table_attribute(3, nullptr);
+    ASSERT_TRUE(status == SAI_STATUS_SUCCESS);
+
+    status = acl_api->get_acl_table_attribute(4, 0, nullptr);
+    ASSERT_TRUE(status == SAI_STATUS_SUCCESS);
+
+    sai_object_id_t exp_oid_1 = 100;
+    sai_object_id_t exp_oid_2 = 200;
+
+    auto x = SpyOn<0, offsetof(sai_acl_api_t, create_acl_table)>(&acl_api.get()->create_acl_table);
+    x->callFake([&](sai_object_id_t* oid, sai_object_id_t, uint32_t, const sai_attribute_t*) -> sai_status_t {
+        *oid = exp_oid_1;
+        return (sai_status_t)SAI_STATUS_SUCCESS;
+    });
+
+    auto y = SpyOn<0, offsetof(sai_acl_api_t, remove_acl_table)>(&acl_api.get()->remove_acl_table);
+    y->callFake([&](sai_object_id_t oid) -> sai_status_t {
+        return (sai_status_t)(oid == exp_oid_2 ? SAI_STATUS_SUCCESS : SAI_STATUS_FAILURE);
+    });
+
+    auto w = SpyOn<0, offsetof(sai_acl_api_t, set_acl_table_attribute)>(&acl_api.get()->set_acl_table_attribute);
+    w->callFake([&](sai_object_id_t oid, const sai_attribute_t*) -> sai_status_t {
+        return (sai_status_t)(oid == exp_oid_2 ? SAI_STATUS_SUCCESS : SAI_STATUS_FAILURE);
+    });
+
+    auto z = SpyOn<0, offsetof(sai_acl_api_t, get_acl_table_attribute)>(&acl_api.get()->get_acl_table_attribute);
+    z->callFake([&](sai_object_id_t oid, uint32_t, sai_attribute_t*) -> sai_status_t {
+        return (sai_status_t)(oid == exp_oid_2 ? SAI_STATUS_SUCCESS : SAI_STATUS_FAILURE);
+    });
+
+    auto x_fake = x->fake;
+    auto x_spy = x->spy;
+    auto x_spy_ref = &x->spy;
+
+    auto y_fake = y->fake;
+    auto y_spy = y->spy;
+    auto y_spy_ref = &y->spy;
+
+    auto w_fake = w->fake;
+    auto w_spy = w->spy;
+    auto w_spy_ref = &w->spy;
+
+    auto z_fake = z->fake;
+    auto z_spy = z->spy;
+    auto z_spy_ref = &z->spy;
+
+    acl_api->create_acl_table(&oid, 1, 0, nullptr);
+    ASSERT_TRUE(oid == exp_oid_1);
```

Comment:
Can you please use ASSERT_EQ? here and below

Reply:
Yes. Will Fix.